pkg/kubectl/configmap.go

```diff
@@ -0,0 +1,204 @@
+func (s ConfigMapGeneratorV1) StructuredGenerate() (runtime.Object, error) {
+	if err := s.validate(); err != nil {
+		return nil, err
+	}
+	configMap := &api.ConfigMap{}
+	configMap.Name = s.Name
+	configMap.Data = map[string]string{}
+	if len(s.FileSources) > 0 {
+		if err := handleConfigMapFromFileSources(configMap, s.FileSources); err != nil {
+			return nil, err
+		}
+	}
+	if len(s.LiteralSources) > 0 {
+		if err := handleConfigMapFromLiteralSources(configMap, s.LiteralSources); err != nil {
+			return nil, err
+		}
+	}
+	return configMap, nil
+}
+
+// validate validates required fields are set to support structured generation
+func (s ConfigMapGeneratorV1) validate() error {
+	if len(s.Name) == 0 {
+		return fmt.Errorf("name must be specified")
+	}
+	return nil
+}
+
+// handleFromLiteralSources adds the specified literal source information into the provided configMap
+func handleConfigMapFromLiteralSources(configMap *api.ConfigMap, literalSources []string) error {
+	for _, literalSource := range literalSources {
+		keyName, value, err := parseLiteralSource(literalSource)
+		if err != nil {
+			return err
+		}
+		err = addKeyFromLiteralToConfigMap(configMap, keyName, value)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// handleFromFileSources adds the specified file source information into the provided configMap
+func handleConfigMapFromFileSources(configMap *api.ConfigMap, fileSources []string) error {
+	for _, fileSource := range fileSources {
+		keyName, filePath, err := parseFileSource(fileSource)
+		if err != nil {
+			return err
+		}
+		info, err := os.Stat(filePath)
+		if err != nil {
+			switch err := err.(type) {
+			case *os.PathError:
+				return fmt.Errorf("error reading %s: %v", filePath, err.Err)
+			default:
+				return fmt.Errorf("error reading %s: %v", filePath, err)
+			}
+		}
+		if info.IsDir() {
+			if strings.Contains(fileSource, "=") {
+				return fmt.Errorf("cannot give a key name for a directory path.")
+			}
+			fileList, err := ioutil.ReadDir(filePath)
+			if err != nil {
+				return fmt.Errorf("error listing files in %s: %v", filePath, err)
+			}
+			for _, item := range fileList {
+				itemPath := path.Join(filePath, item.Name())
+				if item.Mode().IsRegular() {
+					keyName = item.Name()
+					err = addKeyFromFileToConfigMap(configMap, keyName, itemPath)
+					if err != nil {
+						return err
+					}
+				}
+			}
+		} else {
+			err = addKeyFromFileToConfigMap(configMap, keyName, filePath)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func addKeyFromFileToConfigMap(configMap *api.ConfigMap, keyName, filePath string) error {
+	data, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		return err
+	}
+	return addKeyFromLiteralToConfigMap(configMap, keyName, string(data))
+}
+
+func addKeyFromLiteralToConfigMap(configMap *api.ConfigMap, keyName, data string) error {
+	if !validation.IsSecretKey(keyName) {
```

Comment:
@derekwaynecarr Secret and ConfigMap have the same validation requirements for keys